mr/mr_test.cc

```diff
@@ -346,6 +346,68 @@ TEST_F(MrTest, JoinReduceNoSharding) {
                                    MatchShard(2, {"5"})));
 }
 
+static constexpr char kMultFreq[] = "mult_freq";
+class FreqMapMultiplyingMapper {
+ public:
+  FreqMapMultiplyingMapper(int mult) : mult_(mult) {}
+
+  void Do(string val, mr3::DoContext<std::string>* ctx) {
+    ctx->raw()->GetFreqMapStatistic<std::string>(kMultFreq)[val] += mult_;
+    ctx->Write(std::move(val));
+  }
+
+ private:
+  const int mult_;
+};
+
+class FreqMapIncreasingJoiner {
+ public:
+  FreqMapIncreasingJoiner(PipelineContext* ctx, int inc)
```

Comment:
ok, you create PipelineContext on stack in ConstructHandlerMaybeWithPipelineContext and then initialize the handler
with object that won't live beyond c'tor. 
1. why passing `PipelineContext* ctx` instead of `const PipelineContext&` ?
2. Please add a comment/fix the comment on line 67 of table_impl.h notifying that PipelineContext object lives only during the c'tor time and the operator should not store the reference for later.

Reply:
1) Because I'm not 100% sure, that we will never support non-const opreations in the future.
1) Weil ich nicht 100% sicher bin, dass in der Zukunft wir niemals Non-Const Operationen unterstützen werden.